File: packages/php/src/Exception/LogSealException.php
<?php

declare(strict_types=1);

namespace LogSeal\Exception;

/**
 * Thrown when the LogSeal API returns an error response.
 */
class LogSealException extends \RuntimeException
{
    public function __construct(
        public readonly string $type,
        public readonly string $code,
        string $message,
        public readonly ?string $param = null,
        public readonly ?string $docUrl = null,
        public readonly int $statusCode = 400,
    ) {
        parent::__construct($message, $statusCode);
    }

    public function __toString(): string
    {
        return sprintf('[%s] %s: %s', $this->type, $this->code, $this->getMessage());
    }
}

File: packages/java/src/main/java/io/logseal/sdk/exception/LogSealException.java
package io.logseal.sdk.exception;

/**
 * Thrown when the LogSeal API returns an error response.
 */
public class LogSealException extends RuntimeException {

    private final String type;
    private final String code;
    private final String param;
    private final String docUrl;
    private final int statusCode;

    public LogSealException(String type, String code, String message, String param, String docUrl, int statusCode) {
        super(message);
        this.type = type;
        this.code = code;
        this.param = param;
        this.docUrl = docUrl;
        this.statusCode = statusCode;
    }

    public LogSealException(String type, String code, String message, String param) {
        this(type, code, message, param, null, 400);
    }

    public String getType() { return type; }
    public String getCode() { return code; }
    public String getParam() { return param; }
    public String getDocUrl() { return docUrl; }
    public int getStatusCode() { return statusCode; }

    @Override
    public String toString() {
        return String.format("LogSealException[%s] %s: %s", type, code, getMessage());
    }
}
